src/react-disposable-decorator.js

```diff
@@ -9,7 +9,7 @@ if (!enabled) {
 
 export default function(DecoratedComponent) {
 	if (!enabled) {
-		return DecoratedComponent;
+		return <DecoratedComponent cancelWhenUnmounted={() => {}} />;
```

Comment:
Does this work? Before a component is returned. With the change we return a rendered component.